test/ng/cacheFactorySpec.js

```diff
@@ -133,6 +133,16 @@ describe('$cacheFactory', function() {
         expect(cache.info().size).toBe(0);
       }));
 
+      it('should only decrement size when an element is actually removed via remove', inject(function($cacheFactory) {
+        cache.put('foo', 'bar');
+        expect(cache.info().size).toBe(1);
+
+        cache.remove('undefined');
```

Comment:
would it be possible to add a case that the key is `hasOwnProperty` (or some other property that is standard to `Object.prototype`?